src/main/scala/outwatch/dom/helpers/DomUtils.scala

```diff
@@ -172,16 +146,40 @@ object DomUtils {
 
 
   private[outwatch] final case class SeparatedReceivers(
-    childStream: List[ChildStreamReceiver] = Nil,
-    childrenStream: List[ChildrenStreamReceiver] = Nil,
-    attributeStream: List[AttributeStreamReceiver] = Nil
-  )
-  private[outwatch] def separateReceivers(receivers: Seq[Receiver]): SeparatedReceivers = {
-    receivers.foldRight(SeparatedReceivers()) {
-      case (cr: ChildStreamReceiver, sr) => sr.copy(childStream = cr :: sr.childStream)
-      case (cs: ChildrenStreamReceiver, sr) => sr.copy(childrenStream = cs :: sr.childrenStream)
-      case (ar: AttributeStreamReceiver, sr) => sr.copy(attributeStream = ar :: sr.attributeStream)
+    childNodes: List[ChildVNode] = Nil,
+    hasNodeStreams: Boolean = false,
+    attributeStreamReceivers: List[AttributeStreamReceiver] = Nil
+  ) {
+
+    lazy val observable: Observable[(Seq[Attribute], Seq[VNode])] = {
+
+      val childStreamReceivers = if (hasNodeStreams) {
+        childNodes.foldRight(Observable.of(List.empty[VNode])) {
+          case (vn: VNode_, obs) => obs.combineLatestWith(BehaviorSubject(IO.pure(vn)))((nodes, n) => n :: nodes)
```

Comment:
I do not get why we need to include `vn` into the observable here. Isn't it handled separately when we collect the `children` in `ChildrenNodes`?

Reply:
No, those children get displayed just initially. After a `child`/`children` stream element gets emitted, the  parent node gets patched and those children would be removed during the node update if they were not included here. 

In master, the children get preserved during a patch operation, but in the incorrect order. In this PR, they are included in the node change observable in the correct order.

Reply:
I realized that in order for the patch operation to be efficient, we need to generate snabbdom `key`s (if not already present) for all the original children that get included in the node observable.